docs/proposals/nap-waf.md

```diff
@@ -0,0 +1,753 @@
+  namespace: applications
+spec:
+  # Policy source (required due to NAP v5 pre-compilation)
+  policySource:
+    fileLocation: "s3://ngf-waf-policies/production/policy-v1.2.3.tgz"
+    authSecret:
+      name: "policy-store-credentials"
+    validation:
+      checksumVerification: true
+      # Note: Policy content validation handled by NAP v5 components
+
+    # Retry configuration for policy fetch failures
+    retryPolicy:
+      attempts: 3
+      backoff: "exponential"
+      maxDelay: "5m"
+
+    # Timeout for policy downloads
+    timeout: "30s"
+
+  # Security logging configuration for app_protect_security_log directives
+  # Multiple log configurations are supported
+  securityLogs:
+  - name: "stderr-logging"
+    # Built-in logging profile (validated against allowed values)
+    logProfile: "log_all"  # log_all, log_blocked, log_grpc_all, log_grpc_blocked, log_grpc_illegal, log_illegal
+    destination:
+      stderr: true
+
+  - name: "file-logging"
+    # Custom logging profile bundle (similar to policy bundle)
+    # logProfile and logProfileBundle are mutually exclusive per security log configuration entry
+    logProfileBundle:
+      fileLocation: "s3://ngf-waf-policies/logging/custom-log-profile.tgz"
+      authSecret:
+        name: "policy-store-credentials"
+      validation:
+        checksumVerification: true
+      retryPolicy:
+        attempts: 3
+        backoff: "exponential"
+        maxDelay: "5m"
+      timeout: "30s"
+    destination:
+      file:
+        path: "/var/log/app_protect/security.log"
+        # Note: check this folder is actually writeable.
+        # Path must be on shared volume accessible to waf-enforcer container
+
+  - name: "syslog-logging"
+    logProfile: "log_blocked"
+    destination:
+      syslog:
+        server: "syslog-svc.default:514"
+        # Note: TCP transport, unsecured. Use local syslog proxy for secure forwarding
+
+---
+# Status examples showing WafPolicy Conditions
+# TODO - Need to design Conditions
+```
+
+### HTTPRoute Integration
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: protected-application
+spec:
+  parentRefs:
+  - name: nginx-gateway
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: "/api"
+    filters:
+    - type: ExtensionRef
+      extensionRef:
+        group: gateway.nginx.org
+        kind: WafPolicy
+        name: production-waf-policy
+    backendRefs:
+    - name: api-service
+      port: 8080
+```
+
+### Authentication Methods
+
+The policy fetcher supports multiple authentication methods with preference for cloud-native approaches:
+
+**Priority Order:**
+
+1. **Native Cloud Authentication (Recommended)**
+2. **Kubernetes Secrets (Fallback)**
+3. **No Authentication (Public endpoints)**
+
+#### Native Cloud Authentication
+
+For cloud-hosted policy storage, NGF automatically detects and uses native cloud authentication when available. Supported methods:
```

Comment:
Does "automatically detects" mean that no code changes are needed on the NGF side? And we'll be able to support many providers without any updates to our product code?